### DISCUSSION TITLE
Installing MOOSE with OpenMPI3.1.3 and GCC 7

### DISCUSSION THREAD

**Discussion by jinca** [idaholab/moose#16544] [Q&A Getting Started] [answered]
Hello,
I used the following modules:
module purge
module load slurm
module load dot
module load turbovnc/2.0.1
module load vgl/2.5.1/64
module load singularity/current
module load rhel7/global
module load cmake/latest
module load gcc/7
module load openmpi-3.1.3-gcc-7.2.0-b5ihosk

I followed the procedures of the Website:
source ~/.moose_profile
echo $CC
which mpicc
cd
mkdir projects
git clone https://github.com/idaholab/moose
cd moose
git checkout master
cd ~/projects/moose

unset PETSC_DIR PETSC_ARCH
./scripts/update_and_rebuild_petsc.sh --download-mumps=0 --with-64-bit-indices=1

cd petsc/
make PETSC_DIR=/home/ir-inca1/moose/scripts/../petsc PETSC_ARCH=arch-moose check

cd ..
./scripts/update_and_rebuild_libmesh.sh

cd ~/projects/moose/test
make -j 4
./run_tests -j 4

I was able to build PETSC and LibMesh, but when it comes to run the tests, I got the following errors:
postprocessors/num_elems.test_split ............................................ [min_cpus=4] FAILED (CSVDIFF)
relationship_managers/geometric_neighbors.geometric_edge_neighbor .............. [min_cpus=3] FAILED (EXODIFF)
outputs/xml.parallel/distributed ............................................... [min_cpus=3] FAILED (XMLDIFF)
mesh/splitting.use_split ....................................................... [min_cpus=3] FAILED (EXODIFF)
relationship_managers/evaluable.evaluable_neighbors_replicated ................. [min_cpus=3] FAILED (EXODIFF)
relationship_managers/geometric_neighbors.geometric_edge_neighbor_3D ........... [min_cpus=3] FAILED (EXODIFF)
mesh/centroid_partitioner.centroid_partitioner_test ............................ [min_cpus=4] FAILED (EXODIFF)
relationship_managers/evaluable.edge_neighbor .................................. [min_cpus=3] FAILED (EXODIFF)
--------------------------------------------------------------------------------------------------------------
Ran 2416 tests in 772.2 seconds.
2366 passed, 96 skipped, 0 pending, 50 FAILED
MAX FAILURES REACHED

**Comment by aeslaughter** [comment]
There should be some more detailed error information farther up in your terminal window. I suspect there is a problem with the MPI.

**Comment by ngrilli** [comment]
@jinca 
Which operating system are you using?

I suggest you try the manual installation with gcc 9.2.0 and mpich 3.3 on the website
It usually works fine for us

Nicolò

**Reply by jinca** [reply]
The OS is Scientific Linux release 7.9 (Nitrogen)

**Reply by milljm** [reply]
50 Errors (the MAX before it quits) usually means an easy fix. Something fundamentally wrong. Not necessarily a compiler version or the like.

As Andrew asked, seeing the actual error helps tremendously. The details of these errors can be found if you scroll up in your terminal history.

**Comment by jinca** [comment]
OK, When I set the following modules:
module load rhel7/global 
module load slurm           
module load gcc/7
module load openmpi-3.1.3-gcc-7.2.0-b5ihosk
module load python/3.8

I got this error while building petsc:
===============================================================================                                                                                                                     Configuring STRUMPACK with cmake; this may take several minutes                                                                                                                         ===============================================================================                                                                                                                                                                                                                                                                                                             *******************************************************************************
         UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
-------------------------------------------------------------------------------
Error configuring STRUMPACK with cmake
*******************************************************************************

There was an error. Exiting...

**Reply by milljm** [reply]
Without PETSc, you wouldn't (shouldn't) be able to build libMesh. Without libMesh... you wouldn't be able to build MOOSE for an attempt at `./run_test`. I see lots of tests passed as well as reaching the max failure. The only thing I can think of, is that your moose repo was previously built, and has a partial successful build.

Its hard to tell if you copied exactly what you attempted, or if there is an error during your attempt:

```bash
which mpicc
cd
mkdir projects                 # <-------- failed to enter projects before cloning moose
git clone https://github.com/idaholab/moose
cd moose
git checkout master
cd ~/projects/moose            # <-------- not the moose you just cloned. A possible older version of moose?
```

**Comment by jinca** [comment]
Thank you so much for the quick replies guys. Many days ago, I was able to run MOOSE only with 7 FAILS. Now I am trying to reinstalling again, I can not install at least PETSC. I am very aware that I have to build PETSC, then LibMesh and at the end MOOSE.
I will clean and delete all, turn off the machine and do it again in order to send the info today asap. Thanks again.

**Comment by jinca** [comment, marked as answer]
Hi guys, this is the configuration I used:
module purge
module load slurm
module load dot
module load turbovnc/2.0.1
module load vgl/2.5.1/64
module load singularity/current
module load rhel7/global
module load cmake/latest
module load gcc/7
module load openmpi-3.1.3-gcc-7.2.0-b5ihosk
module load python/3.8

The PETSC was built as the libMesh, and therefore MOOSE.
I am attaching the output of the tests I've done recently.
Hope you can help me soon.
Best,
Julita Inca


mesh/checkpoint.test_8 .................................................. [insufficient slots,min_cpus=8] SKIP
mesh/checkpoint.test_8a ................................................. [insufficient slots,min_cpus=8] SKIP
partitioners/random_partitioner.test ......................................................... [min_cpus=4] OK
mesh/nemesis.nemesis_repartitioning_test ..................................................... [min_cpus=4] OK
partitioners/petsc_partitioner.ptscotch_weight_both .......................................... [min_cpus=4] OK
partitioners/petsc_partitioner.parmetis ...................................................... [min_cpus=4] OK
partitioners/petsc_partitioner.parmetis_weight_element ....................................... [min_cpus=4] OK
partitioners/petsc_partitioner.parmetis_weight_side .......................................... [min_cpus=4] OK
partitioners/petsc_partitioner.parmetis_weight_both .......................................... [min_cpus=4] OK
partitioners/petsc_partitioner.parmetis_presplit_mesh ........................................ [min_cpus=2] OK
auxkernels/ghosting_aux.no_algebraic_ghosting ................................................ [min_cpus=4] OK
postprocessors/num_nodes.test_split .......................................................... [min_cpus=4] OK
reporters/mesh_info.info/files ........................................... [min_cpus=2] FAILED (MISSING FILES)
outputs/json/distributed.info/default .................................... [min_cpus=2] FAILED (MISSING FILES)
vectorpostprocessors/csv_reader.tester_fail ..................... [min_cpus=2] FAILED (EXPECTED ERROR MISSING)
misc/exception.parallel_error_residual_transient_non_zero_rank .. [min_cpus=2] FAILED (EXPECTED ERROR MISSING)
misc/exception.parallel_error_jacobian_transient_non_zero_rank ................ [min_cpus=2] FAILED (NO CRASH)
bcs/dmg_periodic.1d ............................................................. [min_cpus=2] FAILED (ERRMSG)
vectorpostprocessors/work_balance.work_balance/replicated ...................... [min_cpus=2] FAILED (CSVDIFF)
outputs/variables.nemesis_hide ................................................. [min_cpus=2] FAILED (EXODIFF)
meshgenerators/distributed_rectilinear/generator.3D_hierarch ................... [min_cpus=4] FAILED (EXODIFF)
meshgenerators/distributed_rectilinear/generator.3d_scomm_out .................. [min_cpus=3] FAILED (EXODIFF)
--------------------------------------------------------------------------------------------------------------
Ran 2490 tests in 586.0 seconds.
2480 passed, 96 skipped, 0 pending, 10 FAILED

**Reply by milljm** [reply]
Wrap three ticks around text when wanting to use a pre-rendered block (instead of one). Somehow, doing that entire thing around one broke GitHub! As I only got most of it, from email.

tick tick tick
```
pre-rendered block
```
tick tick tick

Here is an excerpt of what was sent to me by email:

```
problems/eigen_problem/eigensolvers.nonlinear_power ....................................................... OK
ics/depend_on_uo.ic_depend_on_uo: Working Directory: /home/ir-inca1/projects/moose/test/tests/ics/depend_on_uo
ics/depend_on_uo.ic_depend_on_uo: Running command: mpiexec -n 2 /home/ir-inca1/projects/moose/test/moose_test-opt -i geometric_neighbors_ic.i --error --error-unused --error-override --no-gdb-backtrace
ics/depend_on_uo.ic_depend_on_uo: srun: error: Unable to allocate resources: Invalid account or account/partition combination specified
ics/depend_on_uo.ic_depend_on_uo: srun: error: Unable to allocate resources: Invalid account or account/partition combination specified
ics/depend_on_uo.ic_depend_on_uo: Running exodiff: /home/ir-inca1/projects/moose/framework/contrib/exodiff/exodiff -m -F 1e-10 -t 5.5e-06 /home/ir-inca1/projects/moose/test/tests/ics/depend_on_uo/gold/geometric_neighbors_ic_out.e /home/ir-inca1/projects/moose/test/tests/ics/depend_on_uo/geometric_neighbors_ic_out.e
ics/depend_on_uo.ic_depend_on_uo: ERROR:
ics/depend_on_uo.ic_depend_on_uo: *****************************************************************
ics/depend_on_uo.ic_depend_on_uo: EXODIFF (Version: 2.90) Modified: 2018-02-15
ics/depend_on_uo.ic_depend_on_uo: Authors: Richard Drake, rrdrake@sandia.gov
ics/depend_on_uo.ic_depend_on_uo: Greg Sjaardema, gdsjaar@sandia.gov
ics/depend_on_uo.ic_depend_on_uo: Run on 2021/01/13 00:11:20 GMT
ics/depend_on_uo.ic_depend_on_uo: *****************************************************************
ics/depend_on_uo.ic_depend_on_uo:
ics/depend_on_uo.ic_depend_on_uo: exodiff: ERROR: Couldn't open file "/home/ir-inca1/projects/moose/test/tests/ics/depend_on_uo/geometric_neighbors_ic_out.e".
ics/depend_on_uo.ic_depend_on_uo: Reading first file ...
ics/depend_on_uo.ic_depend_on_uo: Reading second file ...
ics/depend_on_uo.ic_depend_on_uo: exodiff: ERROR: exodiff: ERROR: Couldn't open file "/home/ir-inca1/projects/moose/test/tests/ics/depend_on_uo/geometric_neighbors_ic_out.e". File does not exist.
ics/depend_on_uo.ic_depend_on_uo:
ics/depend_on_uo.ic_depend_on_uo: ################################################################################
ics/depend_on_uo.ic_depend_on_uo: Tester failed, reason: EXODIFF
ics/depend_on_uo.ic_depend_on_uo:
ics/depend_on_uo.ic_depend_on_uo ............................................... [min_cpus=2] FAILED (EXODIFF)
```

The issue is with srun:

```
srun: error: Unable to allocate resources: Invalid account or account/partition combination specified
srun: error: Unable to allocate resources: Invalid account or account/partition combination specified
```
Whatever this is, it prevented the following command from running:
```
mpiexec -n 2 /home/ir-inca1/projects/moose/test/moose_test-opt -i geometric_neighbors_ic.i --error --error-unused --error-override --no-gdb-backtrace
```
Which means, there would be no exodiff file generated, which would spawn the error:
```
exodiff: ERROR: exodiff: ERROR: Couldn't open file "/home/ir-inca1/projects/moose/test/tests/ics/depend_on_uo/geometric_neighbors_ic_out.e". File does not exist.
```

The same is happening with every error posted (`srun: error: Unable to allocate resources: Invalid account or account/partition combination specified`). It would seem something to do with the machine you are running on. Allocating resources incorrectly, something like that.
